src/charm.py

```diff
@@ -669,79 +718,29 @@ def _on_restore_backup_action(self, event: ActionEvent) -> None:
             event.fail(message="Backup not found in S3 bucket.")
             return
         try:
-            if not (
-                self._restore_vault(
-                    snapshot=snapshot,
-                    restore_unseal_keys=event.params.get("unseal-keys", ""),
-                    restore_root_token=event.params.get("root-token", ""),
-                )
-            ):
+            if not (self._restore_vault(snapshot=snapshot)):
                 logger.error("Failed to restore vault.")
                 event.fail(message="Failed to restore vault.")
                 return
         except RuntimeError as e:
             logger.error("Failed to restore vault: %s", e)
             event.fail(message="Failed to restore vault.")
             return
-        event.set_results({"restored": event.params.get("backup-id")})
-
-    def _on_set_unseal_keys_action(self, event: ActionEvent) -> None:
-        """Handle the set-unseal-keys action.
-
-        Updates the initialization secret with the provided unseal keys.
 
-        Args:
-            event: ActionEvent
-        """
-        if not self.unit.is_leader():
-            logger.error("Only leader unit can set unseal keys.")
-            event.fail(message="Only leader unit can set unseal keys.")
-            return
-        if not (vault := self._get_initialized_vault_client()):
-            logger.error("Cannot set unseal keys, vault is not initialized yet.")
-            event.fail(message="Cannot set unseal keys, vault is not initialized yet.")
-            return
-        root_token, current_keys = self._get_initialization_secret()
-        new_keys = event.params.get("unseal-keys", "")
-        if set(new_keys) == set(current_keys):
-            logger.info("Unseal keys are already set to %s", new_keys)
-            event.fail(message="Provided unseal keys are already set.")
-            return
-        self._set_initialization_secret(
-            root_token=root_token,
-            unseal_keys=new_keys,
-        )
-        vault.unseal(unseal_keys=new_keys)
-        event.set_results({"unseal-keys": new_keys})
-
-    def _on_set_root_token_action(self, event: ActionEvent) -> None:
-        """Handle the set-root-token action.
-
-        Updates the initialization secret with the provided root token.
+        try:
+            if self._approle_secret_set():
+                role_id, secret_id = self._get_approle_auth_secret()
+                vault = Vault(
+                    url=self._api_address,
+                    ca_cert_path=self.tls.get_tls_file_path_in_charm(File.CA),
+                )
+                if role_id and secret_id and not vault.authenticate(AppRole(role_id, secret_id)):
+                    self._remove_approle_auth_secret()
+        except Exception as e:
```

Comment:
Do we really want to catch a bare `Exception` here? this seems like a bad idea

Reply:
If we hadn't, we would have missed the http Exception about rewinding the body. But I recognize that catching those is a pandoras box so I can change this into VaultClientError and let pythons normal stack trace passthrough on rare exceptions like that.

Reply:
I think it's actually desirable for our charm to crash when unexpected errors occur, so I much prefer this.